api/src/pcapi/core/offers/api.py

```diff
@@ -1824,3 +1826,67 @@ def update_used_stock_price(stock: models.Stock, new_price: float) -> None:
             event=first_finance_event,
             reason=finance_models.PricingLogReason.CHANGE_AMOUNT,
         )
+
+
+def _is_allocine(provider_id: int) -> bool:
+    allocine_products_provider_id = get_allocine_products_provider_id()
+    allocine_stocks_provider_id = get_provider_by_local_class("AllocineStocks").id
+    return provider_id in (allocine_products_provider_id, allocine_stocks_provider_id)
+
+
+def _update_product_extra_data(product: offers_models.Product, movie: offers_models.Movie) -> None:
+    if not product.extraData:
+        product.extraData = offers_models.OfferExtraData()
+
+    extra_data = movie.extra_data or offers_models.OfferExtraData()
+    if movie.allocine_id:
+        extra_data["allocineId"] = int(movie.allocine_id)
+    if movie.visa:
+        extra_data["visa"] = movie.visa
+
+    product.extraData.update((key, value) for key, value in extra_data.items() if value is not None)  # type: ignore[typeddict-item]
```

Comment:
Suite à mon commentaire précédent, si on avait un objet `allocine_extra_data` en plus, cela rendrait la lecture de la fonction un peu plus simple car dans dans le cas non allocine, on verrait que cette fonction modifier simplement `product.extraData["allocineId"]` et `product.extraData["visa"]`. Actuellement la fonction demande pas mal de concentration parce qu'on change d'abord `extra_data` puis on applique `extra_data` à `product.extraData`

```suggestion
    if movie.allocine_id:
        product.extraData["allocineId"] = int(movie.allocine_id)
    if movie.visa:
        product.extraData["visa"] = movie.visa
    if allocine_extra_data:
        product.extraData.update((key, value) for key, value in allocine_extra_data.items() if value is not None)  # type: ignore[typeddict-item]
```